### ISSUE TITLE
dynamicStall causes solver error: Singular Matrix

### ISSUE THREAD

**Issue by hcOnel** [turbinesFoam/turbinesFoam#277]
Hi,
Cases which runs without any problem blows up when dynamicStall model is activated.
```
        dynamicStall
        {
            active          on;
            dynamicStallModel LeishmanBeddoes;
        }

```
I get the error in pimpleFoam logs:
```
PIMPLE: iteration 1
[0] 
[0] 
[0] --> FOAM FATAL ERROR: 
[0] Singular Matrix
[0] 
[0]     From function void Foam::solve(Foam::scalarSquareMatrix&, Foam::List<T>&) [with Type = double; Foam::scalarSquareMatrix = Foam::SquareMatrix<double>]
[0]     in file /cfd/honel/OpenFOAM/OpenFOAM-5.x/src/OpenFOAM/lnInclude/scalarMatricesTemplates.C at line 69.
[0] 
FOAM parallel run exiting
```
Any ideas what causes the singular matrix error?
Thanks in advance.

**Comment by yi-fluid** [comment]
Hi,
Has the issue you mentioned earlier been resolved? I've recently encountered the same problem and am concerned about how to address it.  

Thanks a lot.

**Comment by petebachant** [comment]
Nope, this has not yet been addressed. If you can post a sample case that reproduces the error that would be a good first step.

**Comment by fodorgiulia** [comment]
Hi, 
I'm trying to simulate a small VAWT with ALM and I get the same error when the dynamic stall is on for TSR >1, it works for TSR<1 and dynamic stall on.

**Comment by petebachant** [comment]
Can you provide case files and OpenFOAM/turbinesFoam versions to reproduce the issue?

**Comment by fodorgiulia** [comment]
[crossFlowTurbine_multiRe.zip](https://github.com/user-attachments/files/27256120/crossFlowTurbine_multiRe.zip)
OpenFOAM version 2412, turbinesFoam version 0.2.1
